### PR TITLE
ab2d 1071 create admin api to clear coverage cache

### PR DESCRIPTION
**Pull request by spathiyil**
Create admin api to clear coverage cache

### Summary

Create admin api to clear coverage cache

### Checklist

- [x] Tests and linting pass

### Security
N/A

### Additional JIRA Tickets (optional)

N/A

### CODE REVIEW

File: api/src/main/java/gov/cms/ab2d/api/controller/AdminAPI.java [hunk 1]
```diff
@@ -92,4 +106,13 @@
     public ResponseEntity<List<PropertiesDTO>> updateProperties(@RequestBody List<PropertiesDTO> propertiesDTOs) {
         return new ResponseEntity<>(propertiesService.updateProperties(propertiesDTOs), null, HttpStatus.OK);
     }
+
+
+    @PostMapping("/coverage/clearCache")
```
Comment (adaykin) [inline api/src/main/java/gov/cms/ab2d/api/controller/AdminAPI.java]: I think this should be DeleteMapping, I can't see anything being created...you can then return an HTTP status of deleted.

Comment (spathiyil) [reply api/src/main/java/gov/cms/ab2d/api/controller/AdminAPI.java]: I thought about that at first. 
But one is not allowed to use a `@RequestBody` with Http DELETE verb.

Comment (adaykin) [reply api/src/main/java/gov/cms/ab2d/api/controller/AdminAPI.java]: Right, you would use the HTTP parameters for the variables, there's examples in BulkDataAccessAPI controller that use {variable} in the URL.

File: common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java [hunk 2]
```diff
@@ -22,4 +23,28 @@
 
     Optional<Coverage> findByContractAndBeneficiaryAndPartDMonth(Contract contract, Beneficiary bene, int month);
 
+
+
+    @Modifying
+    @Query(" DELETE FROM Coverage c  ")
+    int deleteAllInBulk();
```
Comment (dkrylovsb) [inline common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java]: `deleteAll`?

Comment (spathiyil) [reply common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java]: This is the signature for delete all 
```
   void deleteAll();
```

It all does not return the number of rows deleted. I was logging the number of rows deleted for the other calls, so opted to be consistent with that.  But I could change it to the deleteAll() method for this case.

File: common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java [hunk 3]
```diff
@@ -22,4 +23,28 @@
...
+
+    @Modifying
+    @Query(" DELETE FROM Coverage c  " +
+            " WHERE c.partDMonth = :month  ")
+    int deleteInBulk(int month);
+
+    @Modifying
+    @Query(" DELETE FROM Coverage c  " +
+            " WHERE c.contract.id = :contractId  ")
+    int deleteInBulk(Long contractId);
```
Comment (dkrylovsb) [inline common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java]: If I have an integer, say, `5`, then `deleteInBulk(5)` would _appear_ to fit either of these invocation, which might be confusing a bit. `deleteByMonth` and `deleteByContractId` would fix this.

File: common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java [hunk 4]
```diff
@@ -22,4 +23,28 @@
...
+
+
+    @Modifying
+    @Query(" DELETE FROM Coverage c  " +
+            " WHERE c.contract.id = :contractId  " +
+            "   AND c.partDMonth = :month  ")
+    int deleteInBulk(Long contractId, int month);
```
Comment (dkrylovsb) [inline common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java]: ...following up on the above, `deleteByMonthAndContract` would settle this interface.

Comment (spathiyil) [reply common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java]: done

File: common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java (hunk 3 above)
Comment (spathiyil) [reply common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java]: done

File: common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java (hunk 2 above)
Comment (spathiyil) [reply common/src/main/java/gov/cms/ab2d/common/repository/CoverageRepository.java]: done - changed to use `deleteAll()`

**Pull request merged**